tests/test_pythia8.py

```diff
@@ -177,3 +186,32 @@ def test_gp():
 def test_gg():
     evt = run_collision(100 * GeV, "gamma", "gamma")
     assert len(evt) > 2
+
+
+def test_cache_file():
+    evt_kin = CenterOfMass(10 * GeV, "p", "p")
+    m = Pythia8(evt_kin, seed=1, cache=True)
+
+    config_lines = "\n".join(m._config)
+    assert "MultipartonInteractions:reuseInit = 3" in config_lines
+
+    cache_file_path = os.path.join(
+        _cache_base_dir(), "Pythia8", "cache_2212_2212_10000.mpi"
+    )
+    assert os.path.exists(cache_file_path)
+    assert os.path.getsize(cache_file_path) > 0
+
+
+def test_cache_file_heavy_ion():
+    evt_kin = CenterOfMass(2000 * GeV, "p", "He")
+    m = Pythia8(evt_kin, seed=1, cache=True)
+
+    config_lines = "\n".join(m._config)
+    assert "HeavyIon:SasdMpiReuseInit = 3" in config_lines
+    assert "HeavyIon:SigFitReuseInit = 3" in config_lines
+
+    cache_file_path = os.path.join(
+        _cache_base_dir(), "Pythia8", "cache_2212_1000020040_2000000.mpi"
```

Comment:
Some of these generated files are very simple. It would be better to actually compare the contents of the sigfit file by comparing `open(fname,"r").read()` of the newly generated file with concrete content such as
```
1 2000 2000
18.3948 
1.90724 
0.357502 
```

This would allow to track (partially) if cache files need updating after Pythia version updates or so.